lib/slsk-client.js

```diff
@@ -18,28 +19,55 @@ class SlskClient extends EventEmitter {
     debug('Init client')
     server = new Server(cb)
 
-    server.on('connectToPeer', peer => {
+    server.on('connect-to-peer', peer => {
       this.connectToPeer(peer)
     })
+
+    server.on('get-peer-address', peer => {
+      peers[peer.user].setAddress(peer.host, peer.port)
+    })
   }
 
   connectToPeer (peer) {
-    debugPeer(`connectToPeer ${peer.user} ${peer.host} ${peer.port} ${peer.token} ${peer.type}`)
+    debug(`connectToPeer ${peer.user} ${peer.host} ${peer.port} ${peer.token} ${peer.type}`)
 
     if (peer.type === 'F') {
       downloadPeerFile(peer.host, peer.port, peer.token, peer.user, false)
       return
     }
 
-    let peerObj = new Peer(peer)
-    peerObj.on('connected', () => {
-      peers[peer.user] = peerObj
-    })
+    if (peers[peer.user]) {
+      debug(`Already connected to ${peer.user}`)
+    } else {
+      peers[peer.user] = new Peer(net.createConnection({
+        host: peer.host,
+        port: peer.port
+      }), peer)
+      peers[peer.user].on('disconnect', () => {
+        delete peers[peer.user]
+      })
+    }
   }
 
   login (credentials, cb) {
     stack.currentLogin = credentials.user
     server.login(credentials)
+    let waitPort = 2234
```

Comment:
Must be settable by user (if not set by user, don't set it and the server will choose a random one)